src/coprocessor/xeval/builtin_math.rs

```diff
@@ -39,9 +40,13 @@ impl Evaluator {
         }
     }
 
-    pub fn abs_real(&mut self, _ctx: &EvalContext, _expr: &Expr) -> Result<Datum> {
-        // TODO add impl
-        Err(Error::Eval(ERROR_UNIMPLEMENTED.to_owned()))
+    pub fn abs_real(&mut self, ctx: &EvalContext, expr: &Expr) -> Result<Datum> {
+        let child = try!(self.get_one_child(expr));
+        let d = try!(self.eval(ctx, child));
+        match d {
+            Datum::F64(f) => Ok(Datum::F64(f.abs())),
+            _ => invalid_type_error(&d, TYPE_FLOAT),
```

Comment:
It seems we should return `Datum::None` when d is `None` according tidb (https://github.com/pingcap/tidb/blob/master/expression/builtin_math.go#L121)  @XuHuaiyu